File: files/samples/py/Game_BunnyStorm.py
# cfg.Game, cfg.Landscape

import random

bunnys = []
gravity = 1/gfx.height
startBunnyCount = 3
amount = 10
isAdding = False

def OnLoad():
    global wabbitTexture, isAdding
    gfx.SetBackColor(0xffffff)

    wabbitTexture = gfx.CreateTexture("/Sys/Img/Bunny.png")

    gfx.SetOnTouchDown(lambda ev, y: isAdding = True)
    gfx.SetOnTouchUp(lambda ev, y: isAdding = False)

def OnReady():
    for i in range(startBunnyCount):
        AddBunny()

    gfx.Play()

def AddBunny():
    global vx
    bunny = gfx.CreateBasicSprite(wabbitTexture)
    bunny.vx = (random.random()*10) / gfx.width
    bunny.vy = ((random.random()*10)-5) / gfx.height

    bunny.x = bunny.y = 0
    bunnys.append(bunny)

    gfx.AddSprite(bunny)

def OnAnimate():
    if isAdding:
        for i in range(amount):
            AddBunny()

    for i in range(len(bunnys)):
        bunny = bunnys[i]
        bunny.x += bunny.vx
        bunny.y += bunny.vy
        bunny.vy += gravity

        if bunny.x > 1:
            bunny.vx *= -1
            bunny.x = 1
        elif bunny.x < 0:
            bunny.vx *= -1
            bunny.x = 0

        if bunny.y > 1:
            bunny.vy *= -0.85
            bunny.y = 1
            if random.random() > 0.5:
                bunny.vy -= random.random() * 6 / gfx.height
        elif bunny.y < 0:
            bunny.vy = 0
            bunny.y = 0

File: files/samples/js/general/Game_BunnyStorm.js

//Configure app for landscape game mode.
cfg.Game, cfg.Landscape

//Variables
var bunnys = []
var gravity = 1/gfx.height
var startBunnyCount = 3
var amount = 10
var isAdding = false

//Handle game loading.
function OnLoad()
{
	gfx.SetBackColor( 0xffffff )
	
	//Pre-load the bunny texture.
	wabbitTexture = gfx.CreateTexture( "/Sys/Img/Bunny.png" )
    
	//Handle touch events.
	gfx.SetOnTouchDown( function(ev){isAdding = true;} )
	gfx.SetOnTouchUp( function(ev){isAdding = false;} )
}

//Called when game has loaded.
function OnReady()
{
	//Add some bunnies.
	for (var i = 0; i < startBunnyCount; i++) 
	    AddBunny()
	
    //Start animating
	gfx.Play()
}

//Add a bunny to scene.
function AddBunny() 
{
	var bunny = gfx.CreateBasicSprite( wabbitTexture )
	bunny.vx = (Math.random()*10) / gfx.width
	bunny.vy = ((Math.random()*10)-5) / gfx.height
	
	bunny.x = bunny.y = 0
	bunnys.push(bunny)
	
	gfx.AddSprite( bunny )
}

//Update bunny positions.
function OnAnimate()
{
	//Add more bunnies when touched
	if( isAdding )
	{
		for (var i = 0; i < amount; i++) 
			AddBunny()
	}
	
	//Bounce bunnies.
	for (var i = 0; i < bunnys.length; i++) 
	{
		var bunny = bunnys[i]
		bunny.x += bunny.vx
		bunny.y += bunny.vy
		bunny.vy += gravity
	
		if (bunny.x > 1)
		{
			bunny.vx *= -1
			bunny.x = 1
		}
		else if (bunny.x < 0)
		{
			bunny.vx *= -1
			bunny.x = 0
		}
		
		if (bunny.y > 1)
		{
			bunny.vy *= -0.85
			bunny.y = 1
			if (Math.random() > 0.5)
				bunny.vy -= Math.random() * 6 / gfx.height
		} 
		else if (bunny.y < 0)
		{
			bunny.vy = 0;
			bunny.y = 0
		}
	}
}

